scripts/dotnet.py

```diff
@@ -0,0 +1,387 @@
+    with respect to the current working directory.
+    '''
+    if platform == 'win32':
+        pattern = '*/{Configuration}/{TargetFramework}/'.format(
+            Configuration=configuration,
+            TargetFramework=framework
+        )
+    else:
+        pattern = '*{Configuration}/*{TargetFramework}/'.format(
+            Configuration=configuration,
+            TargetFramework=framework
+        )
+
+    for path_name in iglob(pattern, recursive=True):
+        if path.isdir(path_name):
+            return path_name
+
+    raise ValueError(
+        'Unable to determine directory for the specified pattern.')
+
+
+def get_directory() -> str:
+    '''Gets the default directory where dotnet is to be installed.'''
+    return path.join(get_tools_directory(), 'dotnet')
+
+
+def install(
+        architecture: str,
+        channel: str,
+        verbose: bool,
+        install_dir: str = None) -> None:
+    '''
+    Downloads dotnet cli into the tools folder.
+    '''
+    start_msg = "Downloading DotNet Cli"
+    getLogger().info('-' * len(start_msg))
+    getLogger().info(start_msg)
+    getLogger().info('-' * len(start_msg))
+
+    if not install_dir:
+        install_dir = get_directory()
+    if not path.exists(install_dir):
+        makedirs(install_dir)
+
+    getLogger().info("DotNet Install Path: '%s'", install_dir)
+
+    # Download appropriate dotnet install script
+    dotnetInstallScriptExtension = '.ps1' if platform == 'win32' else '.sh'
+    dotnetInstallScriptName = 'dotnet-install' + dotnetInstallScriptExtension
+    url = 'https://raw.githubusercontent.com/dotnet/cli/{}/scripts/obtain/'
+    dotnetInstallScriptUrl = url.format(channel) + dotnetInstallScriptName
+
+    dotnetInstallScriptPath = path.join(install_dir, dotnetInstallScriptName)
+
+    getLogger().info('Downloading %s', dotnetInstallScriptUrl)
+    urlretrieve(dotnetInstallScriptUrl, dotnetInstallScriptPath)
+
+    if platform != 'win32':
+        chmod(dotnetInstallScriptPath, S_IRWXU)
+
+    dotnetInstallInterpreter = [
+        'powershell.exe',
+        '-NoProfile',
+        '-ExecutionPolicy', 'Bypass',
+        dotnetInstallScriptPath
+    ] if platform == 'win32' else [dotnetInstallScriptPath]
+
+    # TODO: Define what is the user story of this script.
+    SDK_CHANNELS = [channel]
+
+    # Install Runtime/SDKs
+    for sdk_channel in SDK_CHANNELS:
+        cmdline_args = dotnetInstallInterpreter + [
+            '-InstallDir', install_dir,
+            '-Architecture', architecture,
+            '-Channel', sdk_channel
+        ]
+        RunCommand(cmdline_args, verbose=verbose).run(
+            get_repo_root_path()
+        )
+
+
+def add_arguments(parser: ArgumentParser) -> ArgumentParser:
+    '''
+    Adds new arguments to the specified ArgumentParser object.
+    '''
+
+    if not isinstance(parser, ArgumentParser):
+        raise TypeError('Invalid parser.')
+
+    SUPPORTED_ARCHITECTURES = [
+        'x64',  # Default architecture
```

Comment:
```cmd
py.exe .\scripts\benchmarks_ci.py --architecture x86 -f netcoreapp3.0 --category CoreCLR
py.exe .\scripts\benchmarks_ci.py --architecture x64 -f netcoreapp3.0 --category CoreCLR
```

Expected: runs as x86, then x64
Actual: runs twice as x86 (it was provided as first)

Reply:
Good catch! I'll fix it!

---
In reply to: [237097443](https://github.com/dotnet/performance/pull/150#discussion_r237097443) [](ancestors = 237097443)